packages/dataviews/src/filter-summary.js

```diff
@@ -62,127 +56,140 @@ const FilterText = ( { activeElement, filterInView, filter } ) => {
+				{ filter.name }
+			</FlexItem>
+			{ operatorOptions.length > 1 && (
+				<SelectControl
+					label={ filter.name }
+					value={ value }
+					options={ operatorOptions }
+					onChange={ ( newValue ) => {
+						const newFilters = currentFilter
+							? [
+									...view.filters.map( ( _filter ) => {
+										if ( _filter.field === filter.field ) {
+											return {
+												..._filter,
+												operator: newValue,
+											};
+										}
+										return _filter;
+									} ),
+							  ]
+							: [
+									...view.filters,
+									{
+										field: filter.field,
+										operator: newValue,
+									},
+							  ];
+						onChangeView( {
+							...view,
+							page: 1,
+							filters: newFilters,
+						} );
+					} }
+					size="small"
+					__nextHasNoMarginBottom
+					hideLabelFromVision
+				/>
+			) }
+		</HStack>
+	);
 }
 
-export default function FilterSummary( { filter, view, onChangeView } ) {
-	const filterInView = view.filters.find( ( f ) => f.field === filter.field );
-	const otherFilters = view.filters.filter(
-		( f ) => f.field !== filter.field
+function ResetFilter( { filter, view, onChangeView, addFilterRef } ) {
+	const isDisabled =
+		view.filters.find( ( _filter ) => _filter.field === filter.field )
+			?.value === undefined && filter.isPrimary;
+	return (
+		<div className="dataviews-filter-summary__reset">
+			<Button
+				disabled={ isDisabled }
+				__experimentalIsFocusable
+				__next40pxDefaultSize
+				variant="tertiary"
+				style={ { justifyContent: 'center', width: '100%' } }
+				onClick={ () => {
+					onChangeView( {
+						...view,
+						page: 1,
+						filters: view.filters.filter(
+							( _filter ) => _filter.field !== filter.field
+						),
+					} );
+					// If the filter is not primary and can be removed, it will be added
+					// back to the available filters from `Add filter` component.
+					if ( ! filter.isPrimary ) {
+						addFilterRef.current?.focus();
+					}
+				} }
+			>
+				{ filter.isPrimary ? __( 'Reset' ) : __( 'Remove' ) }
+			</Button>
+		</div>
 	);
+}
+
+export default function FilterSummary( props ) {
+	const toggleRef = useRef();
+	const { filter, view } = props;
+	const filterToOpenOnMount = useSelect(
+		( select ) => select( dataviewsStore ).getOpenFilterOnMount(),
+		[]
+	);
+	const filterInView = view.filters.find( ( f ) => f.field === filter.field );
 	const activeElement = filter.elements.find(
 		( element ) => element.value === filterInView?.value
 	);
-	const activeOperator = filterInView?.operator || filter.operators[ 0 ];
-
 	return (
-		<DropdownMenu
-			key={ filter.field }
-			trigger={
-				<Button variant="tertiary" size="compact" label={ filter.name }>
+		<Dropdown
+			defaultOpen={ filterToOpenOnMount === filter.field }
+			contentClassName="dataviews-filter-summary__popover"
+			popoverProps={ { placement: 'bottom-start', role: 'dialog' } }
+			onClose={ () => {
+				toggleRef.current?.focus();
```

Comment:
Maybe this could be handled inside DropDown by checking the `defaultOpen` and maybe some more values like `focusOnMount`.  🤔 . The issue is that we mount the DropDown by clicking something that is no longer rendered(through add filter), so if we close it, focus is lost. --cc @mirka